mobile/library/common/jni/jni_helper.h

```diff
@@ -269,8 +271,19 @@ class JniHelper {
    */
   void setObjectArrayElement(jobjectArray array, jsize index, jobject value);
 
-  [[nodiscard]] PrimitiveArrayCriticalUniquePtr getPrimitiveArrayCritical(jarray array,
-                                                                          jboolean* is_copy);
+  /**
+   * Returns the pointer into the primitive array.
+   *
+   * https://docs.oracle.com/en/java/javase/17/docs/specs/jni/functions.html#getprimitivearraycritical-releaseprimitivearraycritical
+   */
+  template <typename T = void*>
```

Comment:
out of curiosity, what does `<typename T = void*>` do?

Reply:
Basically if we don't specify the type parameter `getPrimitiveArrayCritical`, the return type will have type parameter `void *`, but we can also specify the type as `getPrimitiveArrayCritical<jlong*>` and the return type will have type parameter `jlong*`. It's a convenient way without having to do manual casting.

Reply:
Oh got it, it's a way to specify a default template param if none is specified in the call. Thanks, hadn't seen that before (or I did and didn't pay it attention!)